schema/draft/schema.ts

```diff
@@ -220,6 +220,22 @@ export interface ClientCapabilities {
    * Experimental, non-standard capabilities that the client supports.
    */
   experimental?: { [key: string]: object };
+  /**
+   * Optional extensions that the client supports.
+   *
+   * Keys are extension identifiers in the format `{vendor-prefix}/{extension-name}`.
+   * The vendor-prefix follows the same format and reservation rules as prefixes in _meta.
+   * Prefixes containing `modelcontextprotocol` or `mcp` are reserved for MCP use.
```

Comment:
Does this need to be more specific?  Or do we just want to delegate explanation to the `_meta` documentation?

Or perhaps it would be better to factor out "reserved key" documentation, and point both `_meta` and `extensions` at that.